h5py/h5d.pyx

```diff
@@ -430,3 +436,64 @@ cdef class DatasetID(ObjectID):
                 efree(offset)
                 if space_id:
                     H5Sclose(space_id)
+
+    IF HDF5_VERSION >= (1, 10, 2):
+
+        def read_direct_chunk(self, offsets, filter_mask=0x0000, PropID dxpl=None):
+            """ (offsets, uint32_t filter_mask=0x0000, PropID dxpl=None)
+
+            Reads data to a bytes array directly from a chunk at position
+            specified by the offsets argument.
+
+            The `filter_mask` is a bit field of up to 32 values. It specify
+            which filters in the pipeline are used with the chunk.
+            Using the default `0x0000` with use all the filters, while `0xFFFF`
+            will skip all the filters.
+
+            Returns a tuple containing the filter_mask and the bytes data which
+            was used for this data.
+
+            Feature requires: 1.10.2 HDF5
+            """
+
+            cdef hid_t dset_id
+            cdef hid_t dxpl_id
+            cdef hid_t space_id = 0
+            cdef hsize_t *offset = NULL
+            cdef size_t data_size
+            cdef int rank
+            cdef uint32_t filters = 0
+            cdef hsize_t read_chunk_nbytes
+            cdef array.array data = array.array('B')
+
+            if filter_mask is None:
+                # Skip all the filters
+                filters = 0xFFFF
+            else:
+                filters = int(filter_mask)
+
+            dset_id = self.id
+            dxpl_id = pdefault(dxpl)
+            space_id = H5Dget_space(self.id)
+            rank = H5Sget_simple_extent_ndims(space_id)
+
+            if offsets is not None:
```

Comment:
This 'if' should be unnecessary - people shouldn't be passing None.

Reply:
Again, I don't think you need to check for `offsets` being None. The corresponding write function doesn't.